src/Worker.cs

```diff
@@ -31,10 +31,34 @@ public async static Task Main(string[] args)
                 LogLevel.Information,
                 string.Format(PowerShellWorkerStrings.PowerShellWorkerVersion, typeof(Worker).Assembly.GetName().Version));
 
-            WorkerArguments arguments = null;
-            Parser.Default.ParseArguments<WorkerArguments>(args)
-                .WithParsed(ops => arguments = ops)
-                .WithNotParsed(err => Environment.Exit(1));
+            var workerOptions = new WorkerOptions();
+
+            var parser = new Parser(settings => settings.EnableDashDash = true);
+            parser.ParseArguments<WorkerArguments>(args)
+                .WithParsed(workerArgs =>
+                {
+                    workerOptions.WorkerId = workerArgs.FunctionsWorkerId ?? workerArgs.WorkerId;
+                    workerOptions.RequestId = workerArgs.FunctionsRequestId ?? workerArgs.RequestId;
+
+                    if (!string.IsNullOrWhiteSpace(workerArgs.FunctionsUri))
+                    {
+                        try
+                        {
+                            var uri = new Uri(workerArgs.FunctionsUri);
+                            workerOptions.Host = uri.Host;
```

Comment:
_Nit comment unrelated to the scope of this PR change:_
You could also consider adding a Uri property to `WorkerOptions` type and assign the `workerArgs.FunctionsUri` to that. then you can avoid rebuilding the URL string inside the `MessagingStream` method. (Pass the `WorkerOptions` instances to that method)

Reply:
Thank you @kshyju, this is a really good suggestion. We will do this work once we remove support for parsing old command-line arguments. I have opened the following issue to track this work https://github.com/Azure/azure-functions-powershell-worker/issues/992.